storage-rdma/src/main/java/org/apache/crail/storage/rdma/client/RdmaStorageLocalEndpoint.java

```diff
@@ -61,11 +61,10 @@ public RdmaStorageLocalEndpoint(InetSocketAddress datanodeAddr) throws Exception
 		this.address = datanodeAddr;
 		this.bufferMap = new ConcurrentHashMap<Long, CrailBuffer>();
 		this.unsafe = getUnsafe();
-		long lba = 0;
 		for (File dataFile : dataDir.listFiles()) {
-			MappedByteBuffer mappedBuffer = mmap(dataFile);
-			bufferMap.put(lba, OffHeapBuffer.wrap(mappedBuffer));
-			lba += mappedBuffer.capacity();
+			long lba = Long.parseLong(dataFile.getName());
+			OffHeapBuffer offHeapBuffer = OffHeapBuffer.wrap(mmap(dataFile));
```

Comment:
@patrickstuedi can you please indent it properly :)

may be you also don't need these two explicit variables, they are just used once. But I understand for clarity it is fine. 

Also - just that I understand, the file names are enumerated with proper name/offset. Because before you had lba+=mappedBuffer.capacity() logic. 

Otherwise looks good to me.

Reply:
actually, the indentation was issue with my display. Sorry.

Reply:
before the hashtable index was a byte offset, now it's just an int index. We could move back to the byte offset by multiplying the lba with the capacity, and then also multiple the aligneLba with the capacity, but that's unnecessary in my mind.

Reply:
the actual problem with the previous code was that the lba was created based on the ordering provided by the Java directory iterator which of course doesn't respect the ordering of the filenames with integer based names..there no guaranteed ordering at all.